lib/idempo.rb

```diff
@@ -58,6 +59,12 @@ def call(env)
 
       expires_in_seconds = (headers.delete("X-Idempo-Persist-For-Seconds") || @persist_for_seconds).to_i
       if response_may_be_persisted?(status, headers, body)
```

Comment:
Most likely you want to call `to_ary` here, actually - because an Array of strings may be persistable, and replacing `body` with the "materialized" array will allow `response_may_be_persisted?` to size it correctly by a sum of bytesizes of the array elements. So first call `to_ary` to materialize, _then_ check for whether the materialized body is too big.